### ISSUE TITLE
case statement highlighting problems

### ISSUE THREAD

**Issue by Zhengqbbb** [jeff-hykin/better-shell-syntax#41]
## The code with a problem is:  [theme: "Vitesse Theme QB" ]
```sh
aaa(){
    local bbb="Hello"
    printf "%s\n" "Hello World"
    echo "Hello World"

    case "${1}" in
        1)
            local bbb="World"
            echo "$bbb"
            ;;
        2)  local bbb="World";   echo "$bbb" ;;
        3)  printf "%s\n" "$bbb"             ;;
        *)  echo "$bbb"                      ;;
    esac
}
```

## It looks like:

**1. `printf` is builtin function**
<img width="575" alt="image" src="https://user-images.githubusercontent.com/40693636/217747188-415a9736-6815-48a4-9f04-b0c985c525b4.png">

**2. `local` resolve fail on single line case body**
<img width="683" alt="image" src="https://user-images.githubusercontent.com/40693636/217747221-e82cef92-1ea2-4ffd-bdf0-751c080383a0.png">

## It should look like:
** description of ideal **
<img width="545" alt="image" src="https://user-images.githubusercontent.com/40693636/217747408-06c74359-365f-48fe-bb92-ff30e59b1840.png">

<img width="662" alt="image" src="https://user-images.githubusercontent.com/40693636/217747445-fb159480-3850-448e-95a2-1389ceaa7110.png">

**Comment by jeff-hykin** [comment]
Might take a bit for this to get fixed. I'll have to convert the legacy case statement & friends to a modern pattern.
Thanks for reporting it though

**Comment by Zhengqbbb** [comment]
Anyway. Thx you work for SHELL 😎

**Comment by jeff-hykin** [comment]
redid the whole case syntax matching system, so this should be fixed in 1.7.0
